sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClient;
+
+import java.util.List;
+import java.util.Map;
+
+@ServiceClient(
+    builder = TableClientBuilder.class)
+public class TableClient {
+    String tableName;
+
+    TableClient(String tableName) {
+        this.tableName = tableName;
+    }
+
+    /**
+     * returns the table associated with this TableClient
+     *
+     * @param tableName name of the table
+     * @return the AzureTable
+     */
+    public AzureTable getTable(String tableName) {
+        return null;
+    }
+
+    /**
+     * Queries and returns entities in the given table using the select and filter strings
+     *
+     * @param top odata top integer
+     * @param selectString odata select string
+     * @param filterString odata filter string
+     * @return a list of the tables that fit the query
+     */
+    public List<TableEntity> queryEntity(Integer top, String selectString, String filterString) {
+        return null;
+    }
+
+    /**
+     * insert a TableEntity with the given properties and return that TableEntity
+     *
+     * @param row the RowKey
+     * @param partition the PartitionKey
+     * @param tableEntityProperties a map of properties for the TableEntity
+     * @return the created TableEntity
+     */
+    public TableEntity insertEntity(String row, String partition, Map<String, Object> tableEntityProperties) {
+        return new TableEntity();
+    }
+
+    /**
+     * insert a TableEntity with the given row and partition key and return that TableEntity
+     * @param row row key
+     * @param partition partition key
+     * @return the created TableEntity
+     */
+    public TableEntity insertEntity(String row, String partition) {
```

Comment:
I definitely see the value in `insertEntity(TableEntity)`, and I -think- I could be convinced that we should have a convenience method that takes all the properties needed to create a TableEntity (i.e. `insertEntity(row, partition, properties)`), but I'm not clear if having a method that just takes row and partition without any properties, just to insert an empty entity in the table, is actually useful to the consumer.

I think I would start with having just a single `insertEntity(TableEntity)` method, and we can always add more later if UX study indicates it would be useful, but we can't take them away once they ship.